test/unit/sthDatabaseNameMapperTool_test.js

```diff
@@ -354,309 +348,401 @@ function shouldMapOrUnmapCollectionTest(databaseName, collectionName, mappingFla
-         sthDatabaseNaming.getAggregatedCollectionName(COLLECTION_NAME_PARAMS) +
-         ' of database \'' + DATABASE_NAME + '\'',
-         shouldNotMapOrUnmapCollectionTest.bind(
-           null, DATABASE_NAME, sthDatabaseNaming.getAggregatedCollectionName(COLLECTION_NAME_PARAMS),
-           true)
-      );
-
-      after(function() {
-        sthConfig.NAME_MAPPING = ORIGINAL_NAME_MAPPING;
-      });
-      sthConfig.NAME_MAPPING = ORIGINAL_NAME_MAPPING;
+describe('sthDatabaseNameMapperTool tests', function() {
+    var ORIGINAL_NAME_MAPPING = sthConfig.NAME_MAPPING;
+
+    describe('getMappingAnalysis tests', function() {
+        describe('name mapping disabled', function() {
+            sthConfig.NAME_MAPPING = undefined;
+            before(function(done) {
+                sthConfig.NAME_MAPPING = undefined;
+                async.series([connectToDatabase, dropDatabases, createRawAndAggregatedCollections], done);
+            });
+
+            it(
+                'should raise an error if neither -m, --map or -u, --unmap options are set',
+                mapUnmapMandatoryOptionsTest.bind(null, sthDatabaseNameMapperTool.getMappingAnalysis)
+            );
+
+            it(
+                'should raise an error if the -b, --database is not set when the -c, --collection is set',
+                collectionAndDatabaseMandatoryOptionsTest.bind(null, sthDatabaseNameMapperTool.getMappingAnalysis)
+            );
+
+            it("should not detect database '" + DATABASE_NAME + "' as susceptible of being mapped", function(done) {
+                sthDatabaseNameMapperTool.getMappingAnalysis(
+                    {
+                        map: true
+                    },
+                    function(err, analysis) {
+                        if (err) {
+                            return done(err);
+                        }
+                        expect(analysis[DATABASE_NAME]).to.equal(undefined);
+                        done();
+                    }
+                );
+            });
+
+            it(
+                "should not detect database '" + DATABASE_NAME + "' as susceptible of being mapped if filtered out",
+                function(done) {
+                    sthDatabaseNameMapperTool.getMappingAnalysis(
+                        {
+                            map: true,
+                            database: sthConfig.DB_PREFIX + 'inexistent_database'
+                        },
+                        function(err, analysis) {
+                            if (err) {
+                                return done(err);
+                            }
+                            expect(analysis[DATABASE_NAME]).to.be(undefined);
+                            done();
+                        }
+                    );
+                }
+            );
+
+            it(
+                "should not detect collection '" +
+                    sthDatabaseNaming.getRawCollectionName(COLLECTION_NAME_PARAMS) +
```

Comment:
Excesive line break here and other similar places in this file. Please do a general review.

Reply:
Fixed  c00464eb